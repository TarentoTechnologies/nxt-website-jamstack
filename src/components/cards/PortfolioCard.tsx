import { Link } from "gatsby";
import { GatsbyImage, getImage } from "gatsby-plugin-image";
import ReactHtmlParser from "react-html-parser";

import forwardArrow from "../../../static/icons/arrow_forward_black.svg";
import {
  bgCadiumOrange,
  bgGlaucousGreen,
  bgUraniaBlue,
  black87,
  white87,
} from "../../styles/style-guide/ColorGuide.module.css";
import {
  cta1,
  heading700,
  subText2,
  tagLabel,
} from "../../styles/style-guide/Typography.module.css";
import {
  ctaIcon,
  customImgSize,
  customTag,
  heroImage,
  heroProgrammerImage,
  truncateDescription,
} from "./PortfolioCard.module.css";
import { ctaIconPos } from "./ShowcaseCard.module.css";

interface PortfolioCardProps {
  imgSrc?: any;
  imgAlt?: string;
  tag?: string;
  title: string;
  description: any;
  ctaText: string;
  ctaLink: any;
  tagVariant?: string;
  isProgramme?: boolean;
  isCTAExternal?: boolean;
  bgGrey?: boolean;
  descRichText?: boolean;
}

export const PortfolioCard = ({
  imgSrc,
  imgAlt,
  tag,
  title,
  description,
  ctaText,
  ctaLink,
  tagVariant,
  isProgramme,
  isCTAExternal,
  bgGrey = false,
  descRichText = false,
}: PortfolioCardProps) => {
  const image: any = getImage(imgSrc);

  return (
    <div className="container-fluid">
      {/* Image */}
      <div
        className={`${isProgramme ? heroProgrammerImage : heroImage} ${
          bgGrey ? "layoutBg" : "whiteBg"
        } d-flex justify-content-center align-items-center`}
      >
        {/* <GatsbyImage
          image={image}
          alt={imgAlt ? imgAlt : ""}
          className={`${
            isProgramme ? heroProgrammerImage : heroImage
          } ${bgGlaucousGreen} img-fluidlayoutBg ${!imgSrc ? "w-100" : ""}`}
          data-aos="fade-up"
        /> */}
        <GatsbyImage
          image={image}
          alt={imgAlt ? imgAlt : ""}
          className={`img-fluid ${isProgramme ? "w-100" : customImgSize}`}
          data-aos="fade-up"
        />
      </div>

      {/* Tag */}
      {tag && tag !== null && (
        <div
          className={`${customTag} ${
            tagVariant === "blue" ? bgUraniaBlue : bgCadiumOrange
          }  mt-4`}
          data-aos="fade-up"
        >
          <label className={`${tagLabel} ${white87} px-3`}>{tag}</label>
        </div>
      )}

      {/* Heading */}
      <div className="mt-3" data-aos="fade-up">
        <h2 className={`${heading700} ${black87}`}>{title}</h2>
      </div>

      {/* Description */}
      <div
        className={`${subText2} ${black87} ${truncateDescription}`}
        data-aos="fade-up"
      >
        {descRichText
          ? ReactHtmlParser(description?.data?.childMarkdownRemark?.html)
          : description}
      </div>

      {/* CTA */}
      <div className="mt-4" data-aos="fade-up">
        {!isCTAExternal ? (
          <Link to={ctaLink}>
            <label
              className={`${cta1} pe-auto cursorPointer`}
              // role="link"
            >
              {ctaText}
              <span className={`${ctaIconPos}`}>
                <img
                  className={`${ctaIcon} ps-2`}
                  src={forwardArrow}
                  alt={ctaText}
                />
              </span>
            </label>
          </Link>
        ) : (
          <a href={ctaLink} rel="external" target="_blank">
            <label
              className={`${cta1} pe-auto cursorPointer`}
              // role="link"
            >
              {ctaText}
              <span className={`${ctaIconPos}`}>
                <img
                  className={`${ctaIcon} ps-2`}
                  src={forwardArrow}
                  alt={ctaText}
                />
              </span>
            </label>
          </a>
        )}
      </div>
    </div>
  );
};
