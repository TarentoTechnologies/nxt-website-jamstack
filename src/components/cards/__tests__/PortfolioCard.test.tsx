import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { PortfolioCard } from "../PortfolioCard";

const imgData = {
  localFile: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "constrained",
        placeholder: {
          fallback:
            "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAFwABAQEBAAAAAAAAAAAAAAAAAwAEBf/EABUBAQEAAAAAAAAAAAAAAAAAAAEA/9oADAMBAAIQAxAAAAFF5DFvgk//xAAcEAACAgIDAAAAAAAAAAAAAAABAgADBBASExT/2gAIAQEAAQUCS8iG9RPTVrHY8upJ/8QAFREBAQAAAAAAAAAAAAAAAAAAEBH/2gAIAQMBAT8Bh//EABURAQEAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/Aaf/xAAeEAABAgcBAAAAAAAAAAAAAAAAASEQERIjMTJBUf/aAAgBAQAGPwK4kh0b0xCni8NT/8QAGhAAAgMBAQAAAAAAAAAAAAAAAAERITFBkf/aAAgBAQABPyGH9KkuFtuEJCiXgr2zRbTwN7dD/9oADAMBAAIAAwAAABBE7//EABcRAAMBAAAAAAAAAAAAAAAAAAABEUH/2gAIAQMBAT8Qe0U//8QAFxEAAwEAAAAAAAAAAAAAAAAAAAEhUf/aAAgBAgEBPxBbRD//xAAcEAEAAgIDAQAAAAAAAAAAAAABABEhMUFRgXH/2gAIAQEAAT8QtVt1gPzkhRaGsB9uDtDtse+RbVywIyjaBXjEpuXtn//Z",
        },
        images: {
          fallback: {
            src: "/static/eebf97735cd812347673b4702603b08e/c3ed2/482_560x300_8f22c74ee4.png",
            srcSet:
              "/static/eebf97735cd812347673b4702603b08e/b131f/482_560x300_8f22c74ee4.png 140w,\n/static/eebf97735cd812347673b4702603b08e/5618b/482_560x300_8f22c74ee4.png 280w,\n/static/eebf97735cd812347673b4702603b08e/c3ed2/482_560x300_8f22c74ee4.png 560w",
            sizes: "(min-width: 560px) 560px, 100vw",
          },
          sources: [],
        },
        width: 560,
        height: 300,
      },
    },
  },
};

describe("Test if the PortfolioCard component is working as expected", () => {
  it("<PortfolioCard /> matches snapshot", () => {
    const component = render(
      <PortfolioCard
        title={"Spinverse"}
        description={
          "CIBoost is a collective intelligence platform.  A team efficiency enhancement tool that will revolutionize team performance as we know it! Built on a methodology that has been proved to increase performance by up to 2x."
        }
        ctaText={"Read More"}
        ctaLink={"https://www.google.com"}
        tag="Gov"
        imgSrc={imgData}
        tagVariant="orange"
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<PortfolioCard /> renders without exploding", () => {
    const component = render(
      <PortfolioCard
        title={"Spinverse"}
        description={
          "CIBoost is a collective intelligence platform.  A team efficiency enhancement tool that will revolutionize team performance as we know it! Built on a methodology that has been proved to increase performance by up to 2x."
        }
        ctaText={"Read More"}
        ctaLink={"https://www.google.com"}
        tag="Gov"
        imgSrc={imgData}
        tagVariant="orange"
      />
    );
    expect(component.getByTestId("portfolio-card")).toBeInTheDocument();
  });
});
