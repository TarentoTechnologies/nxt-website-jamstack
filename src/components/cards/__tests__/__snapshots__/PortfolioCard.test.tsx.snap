// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the PortfolioCard component is working as expected <PortfolioCard /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid"
    data-testid="portfolio-card"
  >
    <div
      class="heroImage whiteBg d-flex justify-content-center align-items-center"
    />
    <div
      class="customTag bgCadiumOrange  mt-4"
      data-aos="fade-up"
    >
      <label
        class="tagLabel white87 px-3"
      >
        Gov
      </label>
    </div>
    <div
      class="mt-3"
      data-aos="fade-up"
    >
      <h2
        class="heading700 black87"
      >
        Spinverse
      </h2>
    </div>
    <div
      class="subText2 black87 truncateDescription"
      data-aos="fade-up"
    >
      CIBoost is a collective intelligence platform.  A team efficiency enhancement tool that will revolutionize team performance as we know it! Built on a methodology that has been proved to increase performance by up to 2x.
    </div>
    <div
      class="mt-4"
      data-aos="fade-up"
    >
      <a
        href="https://www.google.com"
      >
        <label
          class="cta1 pe-auto cursorPointer"
          role="link"
        >
          Read More
          <span
            class="ctaIconPos"
          >
            <img
              alt="Read More"
              class="ctaIcon ps-2"
              src="test-file-stub"
            />
          </span>
        </label>
      </a>
    </div>
  </div>
</div>
`;
